R/class_all_generics.R

```diff
@@ -0,0 +1,126 @@
+## ----
+#' @title Return URL path
+#'
+#' @description
+#' Returns the Uniform Resource Locator (URL) of a \code{BrapiCon} object.
+#'
+#' @param object an \code{rPHG} local or server connection object.
+#' @param ... Additional arguments, for use in specific methods
+#'
+#' @rdname brapiURL
+#' @export
+setGeneric("brapiURL", function(object, ...) standardGeneric("brapiURL"))
+
+
+## ----
+#' @title Return BrAPI version ID
+#'
+#' @description
+#' Returns the version ID for a BrAPI-compliant PHG server
+#'
+#' @param object an \code{rPHG} local or server connection object
+#' @param ... Additional arguments, for use in specific methods
+#'
+#' @rdname brapiVersion
+#' @export
+setGeneric("brapiVersion", function(object, ...) standardGeneric("brapiVersion"))
+
+
+## ----
+#' @title Return host data
+#'
+#' @description
+#' Returns the host information for a given object
+#'
+#' @param object an \code{rPHG} local or server connection object
+#' @param ... Additional arguments, for use in specific methods
+#'
+#' @rdname host
+#' @export
+setGeneric("host", function(object, ...) standardGeneric("host"))
+
+
+## ----
+#' @title Return protocol value
+#'
+#' @description
+#' Returns the protocol information for a given object
+#'
+#' @param object an \code{rPHG} local or server connection object
+#' @param ... Additional arguments, for use in specific methods
+#'
+#' @rdname httProtocol
+#' @export
+setGeneric("httProtocol", function(object, ...) standardGeneric("httProtocol"))
+
+
+## ----
+#' @title Return type of PHG connection
+#'
```

Comment:
How are phgType and httProtocol different?  If phgType is a connection type, does that mean http vs https?

Reply:
`phgType` is something from v1 since I needed to differentiate between local Postgress/Sqlite connections and the BrAPI connections it returns either `local` or `server` depending on the object created. `httProtocol` simply returns if the server has only http or https encryption. I was thinking maybe further down the road to implement something like this with a direct interface to TileDB.